neuromation/cli/utils.py

```diff
@@ -680,3 +683,163 @@ def pager_maybe(
         click.echo_via_pager(
             itertools.chain(["\n".join(handled)], (f"\n{line}" for line in lines_it))
         )
+
+
+class StyledTextHelper:
+    """ Utilities to perform string operations on terminal styled text with better
+    unicode support.
+
+    NOTE: The helper does not recognise controll sequences other than SGR
+        (Select Graphic Rendition) ones.
+    """
+
+    style_re = re.compile(r"(\033\[(?:\d|;)*m)")
+    ansi_reset_all = "\033[0m"
+    whitespace = "\t\n\x0b\x0c\r "
+
+    @classmethod
+    def is_styled(cls, text: str) -> bool:
+        return "\033" in text
+
+    # As we often do operations on headers or similar static text it makes sense to
+    # cache
+    @classmethod
+    @lru_cache(255)
+    def unstyle(cls, text: str) -> str:
+        return click.unstyle(text)
+
+    @classmethod
+    @lru_cache(255)
```

Comment:
Minor: not sure if the minor speedup worth caching of 255 lines

Reply:
Yea, mostly those operations are quite cheap, let's remove the cache for now.  My initiative was that unstyle is actually called quite a lot on the code and maybe it makes sense to avoid reprocessing it.